keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -0,0 +1,1116 @@
+  // CNIs should report what they see fit in relation to their AdminNetworkPolicy
+  // implementation
+  // +optional 
+  Status   []metav1.Condition
+}
+
+type AdminNetworkPolicySpec struct {
+  // Int value bound to 0 - 999, with 0 signaling the special "baseline" priority
+  Priority      Int 
+  // Allows for multiple AdminNetworkPolicy rules at a single priority level
+  // If empty no rules are applied
+  // A total of 100 total rules will be allowed per each network policy instance, 
+  // this rule count will be calculated as the total summation of the 
+  // AdminNetworkPolicyRules, AdminNetworkPolicyIngressRules, and AdminNetworkPolicyEgressRules
+  // in a single AdminNetworkPolicy Instance. 
+  PolicyRules   []AdminNetworkPolicyRule
+}
+
+type AdminNetworkPolicyRule struct {
+  // No implicit isolation of AppliedTo Pods/Namespaces. 
+  // If no rules are a defined no behavior will be observed 
+  AppliedTo   AppliedTo
+  Ingress     []AdminNetworkPolicyIngressRule
+  Egress      []AdminNetworkPolicyEgressRule
+}
+
+// Exactly one of the selector pointers should be set in a given AppliedTo
+type AppliedTo struct {
+  NamespaceSelector         *metav1.LabelSelector
+  NamespaceAndPodSelector   *NamespaceAndPodSelector
+}
+
+// NamespaceAndPodSelector allows the user to select a namespace/s and optionally 
+// select a given set of pods in that namespace/s
+type NamespaceAndPodSelector struct {
+  NamespaceSelector   metav1.LabelSelector
+  // Used to explicitly select pods within a namespace
+  PodSelector         *metav1.LabelSelector
+}
+
+type AdminNetworkPolicyIngressRule struct {
+  // Name is an identifier for this rule. Must be unique within an instance of ANP. Can be
+  // same across policy instances.
+  // +optional
+  Name      string
+  // Action specifies whether this rule must pass, allow or deny traffic. 
+  // Allow: explicitly allows the selected traffic 
+  // Deny: explicitly denies the selected traffic
+  // Pass: explicitly allows the selected traffic to skip and remaining ANP rules 
+  // and be delegated by K8's Network Policy. 
+  Action    RuleAction
+  // List of ports for incoming traffic.
+  // Each item in this list is combined using a logical OR. If this field is
+  // empty or missing, this rule matches no ports (traffic not restricted by port).
+  // If this field is present and contains at least one item, then this rule 
+  // allows/denies/passes traffic only if the traffic matches at least one port in the list.
+  // +optional
+  Ports     []networkingv1.NetworkPolicyPort
+	// List of sources from which traffic will be allowed/denied/passed to the entities 
+  // selected by this AdminNetworkPolicyRule. Items in this list are combined using a logical OR 
+  // operation. If this field is empty or missing, this rule matches no sources. 
+  // If this field is present and contains at least one item, this rule
+  // allows/denies/passes traffic from the defined AdminNetworkPolicyPeer
```

Comment:
from the defined AdminNetworkPolicyPeer(s)

Reply:
++